presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorTableVersion.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.common.type.Type;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorTableVersion
+{
+    public enum VersionType
+    {
+        TIMESTAMP,
+        VERSION
+    }
+    private final VersionType versionType;
+    private final Type versionExpressionType;
+    private final Object tableVersion;
```

Comment:
As I understand, ``tableVersion`` is always a long value, should we declare it as long?

Reply:
Since we are passing Type, I have kept it generic object. may be we can support other types if needed.